### PR TITLE
Check if subdomain is valid, unique host name

### PR DESCRIPTION
**Pull request by katiejiang**
* Replace application name with application URL in create instance form
* Ensure subdomain is valid (letters, numbers, and dashes)
* Ensure subdomain is unique

Before:
screen shot 2018-02-18 at 6 27 01 pm

After:
screen shot 2018-02-18 at 8 13 41 pm

### CODE REVIEW

**Comment by stephanieyshi** [comment]
We should scope out DB migrations before merging this

File: app/models/instance.py [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Instance(db.Model):
 
     # app_id uniquely identifies a deployment on heroku
     app_id = db.Column(db.String(1000), unique=False)
-    name = db.Column(db.String(64), unique=True)  # heroku namae
```
Comment (HanaPearlman) [inline app/models/instance.py:12]: hahaha

**Comment by HanaPearlman** [comment]
Looks great!

**Comment by katiejiang** [comment]
Wait this doesn't actually change the database lol it's just that I changed a comment in `instance.py` :joy:

**Pull request merged**